### PR TITLE
interpreter: (riscv) gather data from all .data sections

### PR DESCRIPTION
**Pull request by superlopuh**
@xdslproject/risc-v-backend

### CODE REVIEW

File: xdsl/interpreters/riscv.py [hunk 1]
```diff
@@ -281,9 +282,9 @@ def get_data(module_op: ModuleOp) -> dict[str, RawPtr]:
                                 assert (
                                     False
                                 ), f"Unexpected directive {data_op.directive.data}"
-                    return data
-        else:
+        if data is None:
```
Comment (compor) [inline xdsl/interpreters/riscv.py:285]:
```suggestion
       assert data is not None, "Could not find data section"
```

Comment (compor) [reply xdsl/interpreters/riscv.py:285]: I'm a bit confused by all the `if` checks and `assert` statements.

**Review by compor — APPROVED** [review]

File: xdsl/interpreters/riscv.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/interpreters/riscv.py:285]: They're markers of me being lazy, really they should be raising better errors, I'd rather either change them to errors now, or keep the weird if data is None structure.

Comment (superlopuh) [reply xdsl/interpreters/riscv.py]: Swapped asserts

**Review by compor — APPROVED** [review]

**Review by adutilleul — APPROVED** [review]

**Pull request merged**